### PR TITLE
UCT/SELF: fixed potential data corruption

### PR DESCRIPTION
**Pull request by hoopoepg**
- fixed re-entry issue in Self TL
- added helper macro to unregister desc

Fixes https://github.com/openucx/ucx/issues/2200

### CODE REVIEW

File: src/uct/base/uct_iface.h [hunk 1]
```diff
@@ -323,6 +323,13 @@ typedef struct uct_iface_mpool_config {
     }
 
 
+#define UCT_TL_IFACE_PUT_DESC(_desc) \
+    do { \
```
Comment (brminich) [inline src/uct/base/uct_iface.h]: we do not use `do while(0)`, because of coding style

Comment (hoopoepg) [reply src/uct/base/uct_iface.h]: fixed

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5961/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5962/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5965/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5966/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5986/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
@yosefe squash changes?

**Comment by yosefe** [comment]
yes

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5992/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
@yosefe fixed

**Pull request merged**